### PR TITLE
US128609 - Quizzing - Question component UI

### PR DESCRIPTION
**Pull request by maddiegabriel**
[US128609 - [Consistent Evaluation][Quizzing][UI] Add consistent-evaluation-quizzing-question.js](https://rally1.rallydev.com/#/42960320374d/custom/236803223728?detail=%2Fuserstory%2F600466526487)

Story includes:
- question headers
- localized save time
- question score / outof
- autograde/default grade info
- inline htmleditor for feedback

Story does not include:
- connecting question info from the API (used dummy data)
- saving score / feedback info
- showing actual question text / answers - these will be separate components 

question_result_1

question_result_2

### CODE REVIEW

File: components/left-panel/quizzing/consistent-evaluation-evidence-quizzing.js [hunk 1]
```diff
@@ -33,22 +33,22 @@ export class ConsistentEvaluationEvidenceQuizzing extends LocalizeConsistentEval
 
 	render() {
 		// TODO :: conditionally render quizzing-attempt or quizzing-completion-summary based on dropdown
-		return html`
```
Comment (maddiegabriel) [inline components/left-panel/quizzing/consistent-evaluation-evidence-quizzing.js:36]: Commenting out the completion summary rendering so we can test this attempt stuff.

I'll wait to merge until [US128606 - [Consistent Evaluation][Quizzing][UI] Add consistent-evaluation-quizzing-overall-summary.js](https://rally1.rallydev.com/#/42960320374d/custom/236803223728?detail=%2Fuserstory%2F600466036461) is tested

LCB dropdown coming soon!!

File: components/left-panel/quizzing/consistent-evaluation-evidence-quizzing.js [hunk 2]
```diff
@@ -33,22 +33,22 @@ export class ConsistentEvaluationEvidenceQuizzing extends LocalizeConsistentEval
...
-			<d2l-consistent-evaluation-quizzing-overall-summary
-				?skeleton=${this.skeleton}
-				.quizSummaryInfo=${this.quizSummaryInfo}
-				.token=${this.token}
-			></d2l-consistent-evaluation-quizzing-overall-summary>
-		`;
```
Comment (maddiegabriel) [inline components/left-panel/quizzing/consistent-evaluation-evidence-quizzing.js:42]: Commenting out the completion summary rendering so we can test this attempt stuff.

I'll wait to merge until [US128606 - [Consistent Evaluation][Quizzing][UI] Add consistent-evaluation-quizzing-overall-summary.js](https://rally1.rallydev.com/#/42960320374d/custom/236803223728?detail=%2Fuserstory%2F600466036461) is tested

LCB dropdown coming soon!!

File: components/left-panel/quizzing/consistent-evaluation-quizzing-attempt-result.js [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+				attribute: 'feedback',
+				type: String
+			},
+			saveTime: {
+				attribute: 'save-time',
+				type: String
+			},
+			isAutoGraded: {
+				attribute: 'is-auto-graded',
+				type: String
```
Comment (JamieTran) [inline components/left-panel/quizzing/consistent-evaluation-quizzing-attempt-result.js:25]: this feels like it should be a type `Boolean`

File: components/left-panel/quizzing/consistent-evaluation-quizzing-attempt-result.js [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+import { css, html, LitElement } from 'lit-element';
+import { formatTimeSaved } from '../../helpers/submissionsAndFilesHelpers';
+import { GradeType } from '@brightspace-ui-labs/grade-result/src/controller/Grade';
+import { ifDefined } from 'lit-html/directives/if-defined.js';
+import { LocalizeConsistentEvaluation } from '../../../localize-consistent-evaluation.js';
+
+export class ConsistentEvaluationQuizzingAttemptSummary extends LocalizeConsistentEvaluation(LitElement) {
+	static get properties() {
+		return {
+			questionNumber: {
```
Comment (JamieTran) [inline components/left-panel/quizzing/consistent-evaluation-quizzing-attempt-result.js:11]: questionNumber, score, and OutOf should be type `Number`

File: components/left-panel/quizzing/consistent-evaluation-quizzing-attempt-result.js [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+				attribute: 'out-of',
+				type: String
+			}
+		};
+	}
+
+	static get styles() {
+		return [bodyStandardStyles, heading3Styles, labelStyles, css`
+			.d2l-consistent-eval-quiz-question-header {
+				background-color: #dddddd;
```
Comment (JamieTran) [inline components/left-panel/quizzing/consistent-evaluation-quizzing-attempt-result.js:45]: nit: I wonder if there's a colour in https://daylight.d2l.dev/style-elements/colors/ that could replace this.

**Review by JamieTran — APPROVED** [review]
👍

**Review by tdevos-d2l — APPROVED** [review]

File: components/left-panel/quizzing/consistent-evaluation-quizzing-attempt.js [hunk 6]
```diff
@@ -38,14 +40,14 @@ export class ConsistentEvaluationQuizzingAttempt extends LocalizeConsistentEvalu
...
-			${this._renderQuestions()}
+			${this._renderQuestionList()}
 		`;
 	}
 
 	async updated(changedProperties) {
 		super.updated(changedProperties);
 		if (changedProperties.has('quizQuestionList')) {
-			// TODO: figure out how questions will render, then do something like this: this._formatQuizQuestions().then(() => this.requestUpdate());
+			// TODO: figure out how questions will render, then do something like this: this._renderQuestionList().then(() => this.requestUpdate());
```
Comment (Scy-D2L) [inline components/left-panel/quizzing/consistent-evaluation-quizzing-attempt.js:50]: If I'm not mistaken you'll either be getting this for free (the property is tracked, so [if 'updated' recognizes that it's been changed it'll trigger a full render](https://lit-element.polymer-project.org/guide/properties#what-happens-when-properties-change)) or need to handle this elsewhere (if you're adding to / removing from the array by reference, [lit won't pick that up](https://github.com/lit/lit-element/issues/107) and this won't be called) - but that's not an immediate concern. Just leaving this note for later reference :)

File: demo/components/left-panel/quizzing/consistent-evaluation-quizzing-attempt-summary.html [hunk 7]
```diff
@@ -19,43 +19,43 @@ <h2>Quiz Attempt Summary - Recommended Time Limit - not exceeded</h2>
 			<d2l-demo-snippet>
 				<d2l-consistent-evaluation-quizzing-attempt-summary start-time="2020-11-25T15:14:58.800Z"
 					end-time="2020-11-25T16:56:22.800Z" time-spent="3599" time-limit="7200" grace-limit="undefined"
-					exceeded-by="undefined" extended-deadline-exceeded-by="undefined" time-limit-enforced="false">
```
Comment (Scy-D2L) [inline demo/components/left-panel/quizzing/consistent-evaluation-quizzing-attempt-summary.html:22]: nit: these are easier to read and maintain when you write the properties one to a line [like this](https://github.com/BrightspaceHypermediaComponents/consistent-evaluation/blob/master/demo/components/right-panel/consistent-evaluation-right-panel-visual.html#L36)

File: demo/index.html [hunk 8]
```diff
@@ -100,6 +100,13 @@ <h4>Consistent Evaluation Quizzing Attempt Summary</h4>
 				<a href="./components/left-panel/quizzing/consistent-evaluation-quizzing-attempt-summary.html">consistent-evaluation-quizzing-attempt-summary</a>
 			</li>
 		</ul>
+		<h4>Consistent Evaluation Quizzing Attempt Result</h4>
+		<ul>
+			<li>
+				<a
```
Comment (Scy-D2L) [inline demo/index.html:106]: Mysterious two-line `<a>` tag can probably be tidied up ;)

File: test/perceptual/components/left-panel/quizzing/consistent-evaluation-quizzing-attempt-result.visual-diff.html [hunk 9]
```diff
@@ -0,0 +1,67 @@
+<!doctype html>
+<html lang="en">
+	<head>
+		<meta charset="utf-8">
+		<meta name="viewport" content="width=device-width, minimum-scale=1, initial-scale=1, user-scalable=yes">
+		<meta http-equiv="X-UA-Compatible" content="ie=edge">
+		<title>consistent-evaluation-quizzing-attempt-summary visual diff tests</title>
```
Comment (Scy-D2L) [inline test/perceptual/components/left-panel/quizzing/consistent-evaluation-quizzing-attempt-result.visual-diff.html:7]: Might have forgotten to update the component name in the title

File: test/perceptual/components/left-panel/quizzing/consistent-evaluation-quizzing-attempt-summary.visual-diff.html [hunk 10]
```diff
@@ -24,35 +24,35 @@
 		<div id="recommended-not-exceeded">
 			<d2l-consistent-evaluation-quizzing-attempt-summary start-time="2020-11-25T15:14:58.800Z"
 				end-time="2020-11-25T16:56:22.800Z" time-spent="3599" time-limit="7200" grace-limit="undefined"
-				exceeded-by="undefined" extended-deadline-exceeded-by="undefined" time-limit-enforced="false">
```
Comment (Scy-D2L) [inline test/perceptual/components/left-panel/quizzing/consistent-evaluation-quizzing-attempt-summary.visual-diff.html:27]: See previous comment on one attribute per line for readable html

**Review by Scy-D2L — COMMENTED** [review]
Substance lgtm, just suggested some structure changes that could make the code more readable :)

**Review by Scy-D2L — APPROVED** [review]

**Review by rowansilver — APPROVED** [review]

**Pull request merged**